tests/unit/bidi/Button.js

```diff
@@ -0,0 +1,133 @@
+define([
+	"intern!object",
+	"intern/chai!assert"
+], function (registerSuite, assert) {
+	var container, html = "<button is='d-button' id='b1' textDir='ltr'>\u05d0\u05d1\u05d2 ABC</button>" +
+		"<button is='d-button' id='b2' textDir='rtl'>ABC \u05d0\u05d1\u05d2</button>" +
+		"<button is='d-button' id='b3' title='ABC \u05d0\u05d1\u05d2' textDir='auto'>\u05d0\u05d1\u05d2 ABC</button>" +
+		"<button is='d-button' id='b4' label='ABC \u05d2\u05d1\u05d0' textDir='auto'>\u05d0\u05d1\u05d2 ABC</button>";
+
+	registerSuite ({
+		name: "deliteful/Button (bidi)",
+		setup: function () {
+			moduleRequire = require.config({
+				context: "module",
+				baseUrl: "../../../",
+				config: {
+					"requirejs-dplugins/has": {
+						"bidi": true
+					}
+				}
+			});
+		},
+		markup: {
+			beforeEach: function () {
+				container = document.createElement("div");
+				document.body.appendChild(container);
+				container.innerHTML = html;
+			},
+			"ltr": function () {
+				var dfd = this.async();
+				moduleRequire(["delite/register", "deliteful/Button"], dfd.callback(function (register) {
+					register.parse();
+					var b1 = document.getElementById("b1");
+					b1.deliver();
+					assert.strictEqual("\u202a\u05d0\u05d1\u05d2 ABC\u202c", b1.labelNode.textContent, "ltr: wrong displayed value for 'label'");
+					assert.strictEqual("\u202a\u05d0\u05d1\u05d2 ABC\u202c", b1.title, "ltr: wrong default value for 'title'");
+				}));
+			},
+			
+			"rtl": function () {		
+				var dfd = this.async();
+				moduleRequire(["delite/register", "deliteful/Button"], dfd.callback(function (register) {
+					register.parse();
+					var b2 = document.getElementById("b2");
+					b2.deliver();
+					assert.strictEqual("\u202bABC \u05d0\u05d1\u05d2\u202c", b2.labelNode.textContent, "rtl: wrong displayed value for 'label'");
+					assert.strictEqual("\u202bABC \u05d0\u05d1\u05d2\u202c", b2.title, "rtl: wrong default value for 'title'");
+				}));
+			},
+			"auto": function () {
+				var dfd = this.async();
+				moduleRequire(["delite/register", "deliteful/Button"], dfd.callback(function (register) {
+					register.parse();	
+					b3 = document.getElementById("b3");
+					b3.deliver();
+					assert.strictEqual("\u202b\u05d0\u05d1\u05d2 ABC\u202c", b3.labelNode.textContent, "auto: wrong displayed value for 'label'");
+					assert.strictEqual("\u202aABC \u05d0\u05d1\u05d2\u202c", b3.title, "auto: wrong value for 'title'");
+				}));
+			},
+			"auto2": function () {
+				var dfd = this.async();
+				moduleRequire(["delite/register", "deliteful/Button"], dfd.callback(function (register) {
+					register.parse();
+					b4 = document.getElementById("b4");
+					b4.deliver();
+					assert.strictEqual("\u202aABC \u05d2\u05d1\u05d0\u202c", b4.labelNode.textContent, "auto2: wrong displayed value for 'label'");
+					assert.strictEqual("\u202aABC \u05d2\u05d1\u05d0\u202c", b4.title, "auto2: wrong value for 'title'");
+				}));
+			}
```

Comment:
There are a bunch of calls to `register.parse()` that should really be `register.parse(container)`, or something like that.   Only parse on the DOM that you've created for each test run.